src/select/select.js

```diff
@@ -107,82 +96,95 @@ class Select extends React.Component<PropsT, StatelessStateT> {
   ) => {
     const multiple = this.isMultiple();
-    const selected = this.state.selectedOptions.find(
-      tag => tag.id === option.id,
-    );
     let selectedOptions;
-    switch (type) {
-      case STATE_CHANGE_TYPE.select:
-        if (!selected) {
-          selectedOptions = multiple ? this.state.selectedOptions.slice() : [];
-          selectedOptions.push(option);
-          this.setState({selectedOptions: selectedOptions});
-          if (this.props.type === TYPE.select && !multiple) {
-            this.setState({isDropDownOpen: false});
-          }
-          this.props.onChange(e, {type, option, selectedOptions});
-        } else if (multiple) {
-          selectedOptions = this.state.selectedOptions.filter(
-            selectedOption => selectedOption.id !== selected.id,
-          );
-          this.setState({
-            selectedOptions: selectedOptions,
-          });
-          this.props.onChange(e, {
-            type: STATE_CHANGE_TYPE.unselect,
-            option,
-            selectedOptions,
-          });
-        }
-        break;
-      case STATE_CHANGE_TYPE.clearAll: {
-        const selectedOptions = [];
+    if (isClearAll) {
+      // clear all
+      selectedOptions = [];
+      this.setState({selectedOptions: selectedOptions});
+      this.props.onChange(e, {
+        type: STATE_CHANGE_TYPE.unselect,
+        selectedOptions,
+      });
+    } else {
+      const selected = this.state.selectedOptions.find(
+        tag => tag.id === pendingOption.id,
+      );
+      const isSelectOrUnselect = !selected;
+      if (isSelectOrUnselect) {
+        // select
+        selectedOptions = multiple ? this.state.selectedOptions.slice() : [];
+        selectedOptions = selectedOptions.concat([pendingOption]);
         this.setState({selectedOptions: selectedOptions});
-        this.props.onChange(e, {type, selectedOptions});
-        break;
-      }
-      case STATE_CHANGE_TYPE.textChange: {
-        // $FlowFixMe
-        const newTextValue = e.target.value;
-        this.setState({textValue: newTextValue});
-        this.props
-          .onChange(e, {
-            type: STATE_CHANGE_TYPE.textChange,
-            textValue: newTextValue,
-          })
-          .then(() => {
-            this.setState({isDropDownOpen: true});
-            if (this.props.filterable) {
-              let filteredOptions = this.props.options.filter(option =>
-                this.props.filterOption(option, newTextValue),
-              );
-              // reset filtered options for new search
-              if (!filteredOptions.length) {
-                filteredOptions = newTextValue ? [] : null;
-              }
-              this.setState({filteredOptions});
-            }
-          });
-        break;
-      }
-      case STATE_CHANGE_TYPE.keyDown: {
-        this.handledHotKeys(e);
-        break;
+        if (this.props.type === TYPE.select && !multiple) {
+          this.setState({isDropDownOpen: false});
```

Comment:
I'm struggling to think of a use case for giving the user full control over when the dropdown closes. It seems like the logic here may be sufficient for now.

Reply:
right now `isDropDownOpen` sets by many rules, but not from `initialState` from customer. I am not sure that we need to provide it, at least it may be controvertial in Search mode, where no results yet in DropDown

Reply:
It seems like the dropdown won't be closed when an optioned selected and it's a multiselect component.

Reply:
Yes, in multiple options they usually pick many options at once, so if we close it after each selection, that might be not the expected behavior
@nguyensomniac